### ISSUE TITLE
A `GET` on `/sys/pprof/profile` causes a read timeout

### ISSUE THREAD

**Issue by ludovicianul** [hashicorp/vault#18851]
**Describe the bug**
Doing a `GET` on `/sys/pprof/profile` causes a read timeout. 

**To Reproduce**
```shell
curl  -X GET \
      -H "Accept: application/json" -H "Content-Type: application/json" -H "X-Vault-Token: $token" \
      http://localhost:8200/v1/sys/pprof/profile
```

**Expected behavior**
A meaningful HTTP response.

**Environment:**
* Vault Server Version (retrieve with `vault status`): 1.12.2
* Vault CLI Version (retrieve with `vault version`): v1.12.2
* Server Operating System/Architecture: macOS Ventura 13.0.1

**Comment by ryowright** [comment]
I'd like to take a look at this issue if no one is currently working on it.

**Comment by ryowright** [comment]
I tried out the reproduction of the behavior, and I actually did end up receiving data back after 30 seconds as specified on the documentation. I just had to output the data to a file because cURL read it as binary data and did not want to output it to the terminal. Although, there was no HTTP response message or confirmation.

Could the default time parameter be a reason you might be experiencing a read timeout?

![image](https://user-images.githubusercontent.com/43687737/215989216-731afb06-3847-409f-9790-adce6e914441.png)
